metadata-service/restli-servlet-impl/src/main/java/com/linkedin/metadata/resources/entity/EntityResource.java

```diff
@@ -389,36 +417,94 @@ public Task<RollbackResponse> deleteEntities(@ActionParam("registryId") @Optiona
     }, MetricRegistry.name(this.getClass(), "deleteAll"));
   }
 
-  /*
-  Used to delete all data related to an individual urn
+  /**
+   * Deletes all data related to an individual urn(entity).
+   * @param urnStr - the urn of the entity.
+   * @param aspectName - the optional aspect name if only want to delete the aspect (applicable only for timeseries aspects).
+   * @param startTimeMills - the optional start time (applicable only for timeseries aspects).
+   * @param endTimeMillis - the optional end time (applicable only for the timeseries aspects).
+   * @return -  a DeleteEntityResponse object.
+   * @throws URISyntaxException
    */
-  @Action(name = "delete")
+  @Action(name = ACTION_DELETE)
   @Nonnull
   @WithSpan
-  public Task<DeleteEntityResponse> deleteEntity(@ActionParam(PARAM_URN) @Nonnull String urnStr)
-      throws URISyntaxException {
+  public Task<DeleteEntityResponse> deleteEntity(@ActionParam(PARAM_URN) @Nonnull String urnStr,
+      @ActionParam(PARAM_ASPECT_NAME) @Optional String aspectName,
+      @ActionParam(PARAM_START_TIME_MILLIS) @Optional Long startTimeMills,
+      @ActionParam(PARAM_END_TIME_MILLIS) @Optional Long endTimeMillis) throws URISyntaxException {
     Urn urn = Urn.createFromString(urnStr);
     return RestliUtil.toTask(() -> {
+      // Find the timeseries aspects to delete. If aspectName is null, delete all.
+      List<String> timeseriesAspectNames =
+          EntitySpecUtils.getEntityTimeseriesAspectNames(_entityService.getEntityRegistry(), urn.getEntityType());
+      if (aspectName != null && !timeseriesAspectNames.contains(aspectName)) {
+        throw new IllegalArgumentException(String.format("Failed to find timeseries aspect {}", aspectName));
+      }
+      List<String> timeseriesAspectsToDelete =
+          (aspectName == null) ? timeseriesAspectNames : ImmutableList.of(aspectName);
+
       DeleteEntityResponse response = new DeleteEntityResponse();
-      RollbackRunResult result = _entityService.deleteUrn(urn);
+      if (aspectName == null) {
+        RollbackRunResult result = _entityService.deleteUrn(urn);
+        response.setRows(result.getRowsDeletedFromEntityDeletion());
+      }
+      Long numTimeseriesDocsDeleted =
```

Comment:
can we add an if statement to check if timeseriesAspectsToDelete is empty? We shouldn't run timeseries delete if it is empty

Reply:
The called function doesn't do any additional work if this is the case. No need to do additional checks here.